### ISSUE TITLE
Add litescope support to wishbone-util

### ISSUE THREAD

**Issue by mithro** [litex-hub/wishbone-utils#5]

**Comment by mithro** [comment]
Sigrok support might be helped by;
 - https://crates.io/crates/sigrok
 - https://crates.io/crates/sigrok-sys

**Comment by mithro** [comment]
vcd support might be helped by;
 - https://crates.io/crates/vcd

**Comment by xobs** [comment]
The large blocker here is: What is the `litescope` protocol?  What addresses should we peek and poke?

**Comment by mithro** [comment]
Yes, we should write some information on the litescope protocol.

 - https://github.com/mithro/libsigrok/blob/c67b4c11af695e97f9410c1247b02ca06ea30a36/src/hardware/litescope/protocol.c
 - https://github.com/enjoy-digital/litescope/blob/master/litescope/software/driver/analyzer.py
 - https://github.com/enjoy-digital/litescope/blob/master/test/test_analyzer.py

**Comment by xobs** [comment]
The sigrok bindings are ancient -- they were last updated four years ago.  Additionally, they require an external library to be present on the system.  We may be able to modify sigrok-sys such that it statically builds a Rust package from the C source -- much like how we build libusb, and a number of other Rust projects build bindings -- but as it exists right now, it's not very useful.

The vcd crate looks better supported, and was last updated three months ago.

I'm curious how this would actually behave.  What would it look like to create a trigger?  What would the interface look like?

**Comment by mithro** [comment]
I think you just do something like `--trigger=<signalname>:<value> --dump-to output.vcd`?
